apps/explorer/src/app/components/txs/txs-infinite-list.tsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import { FixedSizeList as List } from 'react-window';
+import InfiniteLoader from 'react-window-infinite-loader';
+import { t } from '@vegaprotocol/react-helpers';
+import { TxsInfiniteListItem } from './txs-infinite-list-item';
+import type { ChainExplorerTxResponse } from '../../routes/types/chain-explorer-response';
+
+interface TxsInfiniteListProps {
+  hasMoreTxs: boolean;
+  areTxsLoading: boolean | undefined;
+  txs: ChainExplorerTxResponse[] | undefined;
+  loadMoreTxs: () => void;
+  error: Error | undefined;
+  className?: string;
+}
+
+interface ItemProps {
+  index: number;
+  style: React.CSSProperties;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const NOOP = () => {};
+
+export const TxsInfiniteList = ({
+  hasMoreTxs,
+  areTxsLoading,
+  txs,
+  loadMoreTxs,
+  error,
+  className,
+}: TxsInfiniteListProps) => {
+  if (!txs) {
+    return <div>No items</div>;
+  }
+
+  // If there are more items to be loaded then add an extra row to hold a loading indicator.
+  const itemCount = hasMoreTxs ? txs.length + 1 : txs.length;
+
+  // Pass an empty callback to InfiniteLoader in case it asks us to load more than once.
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  const loadMoreItems = areTxsLoading ? NOOP : loadMoreTxs;
+
+  // Every row is loaded except for our loading indicator row.
+  const isItemLoaded = (index: number) => !hasMoreTxs || index < txs.length;
+
+  const Item = ({ index, style }: ItemProps) => {
+    let content;
+    if (error) {
```

Comment:
question (non-blocking): is this error for the list? do we render it for each item on purpose?

Reply:
It's as per the instructions for react-window-infinite-loader, the list components from react-window don't take error props. I agree it looks odd but I'm happy to rely on the lib's implementation in this case rather than reinventing the wheel.